### PR TITLE
Update Gradle Plugin

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Update the Gradle wrapper and plugin to the most recent versions.  The `variant.generateBuildConfig` and `tasks.copyGradlePropertiesExampleFile.execute()` tasks were updated as well since they are no longer supported.

### Test
- Verify continuous integration build succeeds.
- Verify app builds and runs without error.

### Review
@aforcier and @maxme, I requested your review since you have experience with these files.  @bummytime, I requested you as a reviewer to keep you informed of Simplenote changes.  Only one developer is required to review these changes, but anyone can perform the review.

### CODE REVIEW

File: Simplenote/build.gradle [hunk 1]
```diff
@@ -123,21 +123,16 @@ if(["storeFile", "storePassword", "keyAlias", "keyPassword"].count { !project.ha
...
 task generateCrashlyticsConfig(group: "generate", description: "Generate Crashlytics config") {
     def outputFile = new File("${rootDir}/Simplenote/crashlytics.properties")
     def inputFile = file("${rootDir}/Simplenote/gradle.properties")
     if (!inputFile.exists()) {
-        tasks.copyGradlePropertiesExampleFile.execute()
+        copy {
+            from("${rootDir}/Simplenote")
+            into("${rootDir}/Simplenote")
+            include('gradle.properties-example')
+            rename('gradle.properties-example', 'gradle.properties')
```
Comment (maxme) [inline Simplenote/build.gradle]: It's not new, but we could remove the install step from the travis config:

```
install:
- cp Simplenote/gradle.properties-example Simplenote/gradle.properties
```

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Sun Sep 16 17:38:31 CDT 2018
+#Wed Jun 05 09:47:38 MDT 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-all.zip
```
Comment (maxme) [inline gradle/wrapper/gradle-wrapper.properties]: Should we run `./gradlew wrapper --gradle-version 5.1.1` to update the `.sh` and `.bat` scripts as well?

We could also add the following to the `./build.gradle` file:

```
wrapper {
    distributionType = Wrapper.DistributionType.ALL
}
```

https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper

File: Simplenote/build.gradle (hunk 1 above)
Comment (theck13) [reply Simplenote/build.gradle]: I removed that statement in 605ee7b.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (theck13) [reply gradle/wrapper/gradle-wrapper.properties]: I updated the scripts using the `./gradlew wrapper --gradle-version 5.1.1` command in 071d952.

**Comment by maxme** [comment]
LGTM :shipit:

**Pull request merged**